localstack/utils/aws/aws_responses.py

```diff
@@ -34,18 +36,68 @@ def requests_error_response_json(msg, code=500, error_type='InternalFailure'):
     return flask_to_requests_response(response)
 
 
-def requests_error_response_xml(message, code=400, code_string='InvalidParameter'):
+def requests_error_response_xml(service, message, code=400, code_string='InvalidParameter'):
     response = RequestsResponse()
-    response._content = """<ErrorResponse xmlns="http://sns.amazonaws.com/doc/2010-03-31/"><Error>
+    response._content = """<ErrorResponse xmlns="http://{service}.amazonaws.com/doc/2010-03-31/"><Error>
         <Type>Sender</Type>
         <Code>{code_string}</Code>
         <Message>{message}</Message>
         </Error><RequestId>{req_id}</RequestId>
-        </ErrorResponse>""".format(message=message, code_string=code_string, req_id=short_uid())
+        </ErrorResponse>""".format(service=service, message=message, code_string=code_string, req_id=short_uid())
     response.status_code = code
     return response
 
 
+def requests_error_response_xml_presign_url_auth(message, string_to_sign=None, signature=None, expires=None, code=400,
+        code_string='AccessDenied', aws_access_token='temp'):
+    response = RequestsResponse()
+    if signature and string_to_sign or code_string == 'SignatureDoesNotMatch':
```

Comment:
Would be great if we can try to reduce the duplication in this function a bit (progressively building the final content string, based on which inputs are provided).

Reply:
I have updated this. Now I am converting main xml template to `dict`  type and adding required parameters on runtime.